scripts/download-overrides.py

```diff
@@ -20,34 +20,46 @@ def get_NVRA_from_NEVRA(string: str) -> str:
 
     # return the first hawkey.NEVRA item in the list of possibilities
     rpminfo = nevras[0]
-
-    # come up with rpm NVRA
-    nvra = f"{rpminfo.name}-{rpminfo.version}-{rpminfo.release}.{rpminfo.arch}"
-    return nvra
-
+    return rpminfo
 
 def is_override_lockfile(filename: str) -> bool:
     return (filename.startswith('manifest-lock.overrides.')
             and filename[-4:] in ['json', 'yaml'])
 
+def assert_epochs_match(overrides_epoch: int, rpmfile_epoch: str):
+    # normalize the input into a string
+    if overrides_epoch is None:
+       normalized_overrides_epoch = '(none)'
+    else:
+       normalized_overrides_epoch = str(overrides_epoch)
+    if  normalized_overrides_epoch != rpmfile_epoch:
+        raise Exception(f"Epoch mismatch between downloaded rpm ({rpmfile_epoch})"
+                        f" and overrides file entry ({overrides_epoch})")
 
 assert os.path.isdir("builds"), "Missing builds/ dir; is this a cosa workdir?"
 
 rpms = set()
+os.makedirs('overrides/rpm', exist_ok=True)
 for filename in os.listdir(os.path.join("src/config")):
     if is_override_lockfile(filename):
         with open(f'src/config/{filename}') as f:
             lockfile = yaml.safe_load(f)
         for pkg, pkgobj in lockfile['packages'].items():
-            rpms.add(get_NVRA_from_NEVRA(f"{pkg}-{pkgobj['evra']}"))
+            rpminfo = get_rpminfo(f"{pkg}-{pkgobj['evra']}")
+            rpmnvra = f"{rpminfo.name}-{rpminfo.version}-{rpminfo.release}.{rpminfo.arch}"
+            rpms.add(rpmnvra)
+            subprocess.check_call(['koji', 'download-build', '--rpm', rpmnvra], cwd='overrides/rpm')
```

Comment:
Hmm, if we used the Koji API directly we could verify the epoch before having to download the RPM. But meh... we're only continuously doing this in CI right now and not developers' laptops, so we can do that as a follow up!

Reply:
+1 for follow up